src/libtorch.cc

```diff
@@ -1265,6 +1284,96 @@ ModelInstanceState::Execute(
   }
 }
 
+TRITONSERVER_Error*
+ModelInstanceState::GetNamingConvention(
+    NamingConvention* naming_convention,
+    const std::set<std::string>& allowed_ios)
+{
+  // Rules for (non-Dictionary) input tensor names:
+  // 1. Must be in 'allowed_inputs' (arguments in the forward function)
+  // 2. Must follow the naming convention i.e. <name>__<index>
+  // 3. If neither of the above conditions are satisfied, enforce strict
+  // ordering of model inputs.
+  //
+  // Rules for output tensor names:
+  // 1. Must follow the naming convention i.e. <name>__<index>
+  // 2. If not, we enforce strict ordering of model outputs.
+  std::string deliminator = "__";
+  size_t io_index = 0;
+  std::string io_kind = "input";
+  *naming_convention = NamingConvention::FORWARD_ARGUEMENT;
+  if (allowed_ios.size() == 0) {
+    // symbolizes output
+    io_kind = "output";
+    *naming_convention = NamingConvention::NAMED_INDEX;
+  }
+
+  triton::common::TritonJson::Value ios;
+  RETURN_IF_ERROR(
+      model_state_->ModelConfig().MemberAsArray(io_kind.c_str(), &ios));
+
+  if (io_kind == "input") {
+    for (size_t i = 0; i < ios.ArraySize(); i++) {
+      triton::common::TritonJson::Value io;
+      RETURN_IF_ERROR(ios.IndexAsObject(i, &io));
+
+      // Validate name
+      std::string io_name;
+      RETURN_IF_ERROR(io.MemberAsString("name", &io_name));
+      auto itr = allowed_ios.find(io_name);
+      if (itr == allowed_ios.end()) {
+        *naming_convention = NamingConvention::NAMED_INDEX;
+        break;
+      }
+    }
+  }
+
+  // If not, check if inputs follow INDEX
+  if (*naming_convention == NamingConvention::NAMED_INDEX) {
+    for (size_t i = 0; i < ios.ArraySize(); i++) {
+      triton::common::TritonJson::Value io;
+      RETURN_IF_ERROR(ios.IndexAsObject(i, &io));
+
+      // Validate name
+      std::string io_name;
+      RETURN_IF_ERROR(io.MemberAsString("name", &io_name));
+      try {
+        int start_pos = io_name.find(deliminator);
+        if (start_pos == -1) {
+          throw std::invalid_argument(
+              "input/output must follow naming convention");
+        }
+        // Use atoi to check if the index part of the name is an integer
+        std::atoi(io_name.substr(start_pos + 2).c_str());
```

Comment:
I don't think I understand how this line checks the index. You call std::atoi but don't use the returned value. From [the documentation](https://en.cppreference.com/w/cpp/string/byte/atoi) it doesn't appear to throw an exception either.

Reply:
Not needed anymore since atoi is not needed here

Reply:
PS: It appears testing this that the exception is indeed caught even if no value is returned

Reply:
This appears to be part of the undefined behavior then, or is undocumented: [No-throw guarantee: this function never throws exceptions.](https://www.cplusplus.com/reference/cstdlib/atoi/). Either way, not a good idea when compiling for different systems :)